eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Epoch.hs

```diff
@@ -162,6 +164,15 @@ returnProposalDeposits removedProposals =
   where
     updateUMap = returnProposalDepositsUMap removedProposals
 
+-- | When there have been zero governance proposals to vote on in the previous epoch
+-- increase the dormant-epoch counter by one.
+updateNumDormantEpochs :: GovState era ~ ConwayGovState era => LedgerState era -> LedgerState era
```

Comment:
Could this be replaced with `ConwayEraGov`?
```suggestion
updateNumDormantEpochs :: ConwayEraGov era => LedgerState era -> LedgerState era
```

Reply:
Nope, since `ConwayEraGov` does not cover this constraint. We essentially need this constraint to be able to use `utxosGovStateL`, as `GovState` changes since `Shelley`.

Reply:
that doesn't sound right, `utxosGovState` just returns the `GovState` of the current era. As long as you use lenses that come from the `*EraGov` classes then it shouldn't be necessary to constrain the `GovState` to a concrete datatype.

Reply:
I think the constraint comes from `cgGovSnapshotsL`, which is specific to ConwayGovState. I think it's better to add this lens to the `ConwayEraGov` typeclass and use it from there.

Reply:
> I think it's better to add this lens as a method of the ConwayEraGov typeclass and use it from there.

That's an awesome idea!